crates/holochain/src/conductor/conductor.rs

```diff
@@ -2963,6 +2962,92 @@ mod misc_impls {
+            let mut hash_map: HashMap<CellId, Vec<CapGrantInfo>> = HashMap::new();
+            let grant_query: ChainQueryFilter = ChainQueryFilter::new()
+                .include_entries(true)
+                .entry_type(EntryType::CapGrant);
+            let delete_query: ChainQueryFilter = ChainQueryFilter::new()
+                .include_entries(true)
+                .action_type(ActionType::Delete);
+
+            for cell_id in cell_set.iter() {
+                // create a source chain read to query for the cap grant
+                let chain = SourceChainRead::new(
+                    self.get_or_create_authored_db(
+                        cell_id.dna_hash(),
+                        cell_id.agent_pubkey().clone(),
+                    )?
+                    .into(),
+                    self.get_or_create_dht_db(cell_id.dna_hash())?.into(),
+                    self.get_or_create_space(cell_id.dna_hash())?
+                        .dht_query_cache,
+                    self.keystore().clone(),
+                    cell_id.agent_pubkey().clone(),
+                )
+                .await?;
+
+                // query for the cap grant and delete actions (capability revokes)
+                let grant_list = chain.query(grant_query.clone()).await?;
+                // No cap grants for this cell
+                if grant_list.is_empty() {
+                    continue;
+                }
+                let delete_action_hash_map: HashMap<ActionHash, Timestamp> = chain
+                    .query(delete_query.clone())
+                    .await?
+                    .iter()
+                    .filter_map(|record| {
+                        if let Action::Delete(delete) = record.action() {
+                            Some((delete.deletes_address.clone(), delete.timestamp))
+                        } else {
+                            None
+                        }
+                    })
+                    .collect::<HashMap<ActionHash, Timestamp>>();
+
+                tracing::info!("cap grant revocation list: {:?}", delete_action_hash_map);
+
+                // create a list of CapGrantInfo structs for each cell
+                let mut cap_grants: Vec<CapGrantInfo> = vec![];
+                for grant_record in grant_list {
+                    let cap_action_hash = grant_record.action_hash().clone();
+                    let mut revoke_time: Option<Timestamp> = None;
+                    // set revoke time if delete action exists
+                    if delete_action_hash_map.contains_key(&cap_action_hash) {
+                        // skip grant info if include_revoked is false
+                        if !include_revoked {
+                            continue;
+                        }
+                        revoke_time = delete_action_hash_map
+                            .get(&cap_action_hash)
+                            .map(|time| time.to_owned())
+                    }
```

Comment:
No need to check for deletes if revoked is false.
```suggestion
                // skip grant info if include_revoked is false
                        if !include_revoked {
                            continue;
                        }
                    if delete_action_hash_map.contains_key(&cap_action_hash) {
                        revoke_time = delete_action_hash_map
                            .get(&cap_action_hash)
                            .map(|time| time.to_owned())
                    }
```